protostar/protostar_cli.py

```diff
@@ -185,7 +192,14 @@ async def run(self, args: Any) -> None:
         except ProtostarException as err:
             if err.details:
                 print(err.details)
-            logger.error(err.message)
+            assert self.logger is not None
+            self.logger.error(err.message)
             sys.exit(1)
         except KeyboardInterrupt:
             sys.exit(1)
+
+    def __del__(self):
+        if self.logger is not None:
```

Comment:
I think use of destuctor is a bit awkward, there is not that much control when it runs. What if we introduced an method and called it explicitly from main function after `asyncio.run(protostar_cli.run(parser.parse()))` finishes? Is it possible?